# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I selected the Refresh / Content Opportunity Scoring lane because businesses often have thousands of content pages but limited time and resources to update them. Instead of reviewing every page manually, this project aims to identify which pages should be prioritized for review based on search performance, engagement, and content signals. This lane has a clear business value because it helps content teams make better decisions about where to focus their efforts first.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

This project helps content managers decide which content pages should be refreshed or updated first. Based on the recommendations, the content team can prioritize pages that are more likely to benefit from an update instead of reviewing thousands of pages manually. If the recommendation is wrong, the team may spend time updating pages that do not need attention or miss pages that actually require improvement, reducing efficiency and delaying potential traffic improvements. Machine learning helps by analyzing patterns across many pages and providing decision support for prioritization rather than replacing human judgment.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)

print("Working directory:", os.getcwd())

Working directory: /content/flyrank-ml-internship-starter


In [5]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Dataset shape:", df.shape)
print("Number of clients:", df["client_id"].nunique())
print("\nTrend direction counts:")
print(df["trend_direction"].value_counts())

Dataset shape: (30000, 44)
Number of clients: 32

Trend direction counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


The starter dataset contains 30,000 content pages from 32 pseudonymized clients. Among these pages, 16,262 are labeled as "down", indicating that more than half of the pages experienced a significant decline in impressions during the comparison period. These numbers suggest that content decline is a common challenge, making the Refresh / Content Opportunity Scoring lane a meaningful decision-support problem. Instead of manually reviewing thousands of pages, a machine learning model can help prioritize which pages should be reviewed first.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project provides decision support by identifying content pages that may be good candidates for review based on historical performance and content signals. The recommendations are directional, meaning they help prioritize pages rather than guarantee future outcomes. The model can identify patterns observed in the available data, but it cannot prove that refreshing a page will improve rankings or traffic. It also does not predict Google's ranking algorithm. Final decisions should always be made by the content team using both the model's recommendations and human judgment.

## Self-check

Before you submit, confirm each line honestly:

- [X ] Every section above is filled — markdown thinking AND the code that backs it
- [ X] The notebook runs top to bottom with no errors (Runtime → Run all)
- [X ] No client names, URLs, or private queries anywhere
- [X ] My claims use careful words: observed, measured, directional, decision-support
- [X ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.